os/arch/arm/src/common/up_mpuinit.c

```diff
@@ -50,11 +50,119 @@
  ****************************************************************************/
+static struct mpu_usages_s g_mpu_region_usages;
+
+#ifdef CONFIG_ARMV7M_MPU
+#define MPU_MAX_NREGION		CONFIG_ARMV7M_MPU_NREGIONS
+#elif defined(CONFIG_ARMV8M_MPU)
+#define MPU_MAX_NREGION		CONFIG_ARMV8M_MPU_NREGIONS
+#endif
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: mpu_regs_initialize
+ *
+ * Description:
+ *   Initialize the MPU region numbers.
+ *
+ ****************************************************************************/
+
+void mpu_regs_initialize(FAR struct mpu_usages_s *mpu)
+{
+
+	uint8_t offset = mpu->nregion_board_specific;
+
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	offset += 3;
+#else
+	offset += 1;
+#endif
+	mpu->nregion_common_bin = offset;
+#endif
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	offset += 3;
+#else
+	offset += 1;
+#endif
+	mpu->nregion_app_bin = offset;
+#endif
+
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+	mpu->nregion_stackovf = mpu->nregion_app_bin + 1;
+	mpu->max_nregion = mpu->nregion_stackovf;
+#else
+	mpu->max_nregion = mpu->nregion_app_bin;
+#endif
+}
+
+/****************************************************************************
+ * Name: mpu_get_region_info
+ *
+ * Description:
+ *   Get the MPU region numbers.
+ *
+ ****************************************************************************/
+
+uint8_t mpu_get_region_info(enum mpu_region_usages usages)
+{
+	uint8_t nregion;
+
+	switch (usages) {
+	case MPU_REGION_BOARD_SPECIFIC:
+		nregion = g_mpu_region_usages.nregion_board_specific;
+		break;
+	case MPU_REGION_COMMON_BIN:
+		nregion = g_mpu_region_usages.nregion_common_bin;
+		break;
+	case MPU_REGION_APP_BIN:
+		nregion = g_mpu_region_usages.nregion_app_bin;
+		break;
+	case MPU_REGION_STACKOVF:
+		nregion = g_mpu_region_usages.nregion_stackovf;
+		break;
+	case MPU_REGION_MAX:
+		nregion = g_mpu_region_usages.max_nregion;
+		break;
+	}
+
+	return nregion;
+}
+
+/****************************************************************************
+ * Name: mpu_set_region_info
+ *
+ * Description:
+ *   Set the MPU region numbers.
+ *
+ ****************************************************************************/
+
+void mpu_set_region_info(enum mpu_region_usages usages, uint8_t val)
+{
+	switch (usages) {
+	case MPU_REGION_BOARD_SPECIFIC:
+		g_mpu_region_usages.nregion_board_specific = val;
+		break;
+	case MPU_REGION_COMMON_BIN:
+		g_mpu_region_usages.nregion_common_bin = val;
+		break;
+	case MPU_REGION_APP_BIN:
+		g_mpu_region_usages.nregion_app_bin = val;
+		break;
+	case MPU_REGION_STACKOVF:
+		g_mpu_region_usages.nregion_stackovf = val;
+		break;
+	case MPU_REGION_MAX:
+		g_mpu_region_usages.max_nregion = val;
+		break;
```

Comment:
We set almost region_usages in this file except board specific. I think no need to support others.